### PR TITLE
feat: remove login session cookie

### PR DESCRIPTION
**Pull request by hperl**
## Checklist

- [ ] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md).
- [ ] I have referenced an issue containing the design document if my change
      introduces a new feature.
- [ ] I am following the
      [contributing code guidelines](../blob/master/CONTRIBUTING.md#contributing-code).
- [ ] I have read the [security policy](../security/policy).
- [ ] I confirm that this pull request does not address a security
      vulnerability. If this pull request addresses a security vulnerability, I
      confirm that I got the approval (please contact
      [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push
      the changes.
- [ ] I have added tests that prove my fix is effective or that my feature
      works.
- [ ] I have added or changed [the documentation](https://github.com/ory/docs).

### CODE REVIEW

File: persistence/sql/persister_consent.go [hunk 1]
```diff
@@ -777,3 +785,20 @@ func (p *Persister) FlushInactiveLoginConsentRequests(ctx context.Context, notAf
 
 	return nil
 }
+
+func (p *Persister) mySQLConfirmLoginSession(ctx context.Context, session *flow.LoginSession) error {
+	err := p.Connection(ctx).Create(session)
+	if err != nil {
```
Comment (aeneasr) [inline persistence/sql/persister_consent.go:791]: Shouldn't we check for UniqueViolation instead?

**Review by aeneasr — COMMENTED** [review]
First pass - this looks good. Let's see what the CI has to say!

I assume the cookie is still going to be set after the consent challenge has been accepted, correct? It's simply not needed during the login/consent dance (which was previously the case)

File: persistence/sql/persister_consent.go (hunk 1 above)
Comment (hperl) [reply persistence/sql/persister_consent.go:802]: fixed

**Comment by hperl** [comment]
> First pass - this looks good. Let's see what the CI has to say!

The CI is very excited about this change ;)

> I assume the cookie is still going to be set after the consent challenge has been accepted, correct? It's simply not needed during the login/consent dance (which was previously the case)

Yes, the final session cookie gets written here:

https://github.com/ory/hydra/blob/e2922c4b1aa819561784f247a8d25fff05ad8f0a/consent/strategy_default.go#L480-L492

**Pull request merged**